map/src/menu/MainMenu.js

```diff
@@ -0,0 +1,328 @@
+}) {
+    const ctx = useContext(AppContext);
+
+    const handleDrawer = () => {
+        setOpenMainMenu(!openMainMenu);
+        setMenuInfo(null);
+    };
+
+    const navigate = useNavigate();
+    const openLogin = () => {
+        navigate('/map/loginForm' + window.location.search + window.location.hash);
+    };
+
+    const items = [
+        {
+            name: 'Weather',
+            icon: WeatherIcon,
+            component: <Weather />,
+            type: OBJECT_TYPE_WEATHER,
+            show: true,
+        },
+        {
+            name: 'Tracks',
+            icon: TracksIcon,
+            component: <TracksMenu />,
+            type: OBJECT_TYPE_LOCAL_TRACK || OBJECT_TYPE_CLOUD_TRACK,
+            show: true,
+        },
+        {
+            name: 'Favorites',
+            icon: FavoritesIcon,
+            component: <FavoritesMenu />,
+            type: OBJECT_TYPE_FAVORITE,
+            show: ctx.favorites?.groups?.length > 0,
+        },
+        {
+            name: 'Navigation',
+            icon: NavigationIcon,
+            component: <RouteMenu />,
+            type: OBJECT_TYPE_ROUTE_TRACK,
+            show: true,
+        },
+        {
+            name: 'Map Style',
+            icon: Map,
+            component: <MapStyle />,
+            type: null,
+            show: ctx.develFeatures,
+        },
+        {
+            name: 'Plan a route',
+            icon: PlanRouteIcon,
+            component: <PlanRouteMenu />,
+            type: null,
+            show: true,
+        },
+    ];
+
+    //open main menu if infoblock was opened
+    useEffect(() => {
+        if (showInfoBlock && !menuInfo) {
+            setOpenMainMenu(true);
+            selectMenuInfo();
+        }
+    }, [showInfoBlock]);
+
+    //open main menu if currentObjectType was changed
+    useEffect(() => {
+        if (ctx.currentObjectType) {
+            if (menuInfo?.type.name !== ctx.currentObjectType) {
+                selectMenuInfo();
+            }
+        }
+    }, [ctx.currentObjectType]);
+
+    function selectMenuInfo() {
+        const currentMenu = items.find((item) => item.type === ctx.currentObjectType);
+        if (currentMenu) {
+            setMenuInfo(currentMenu.component);
+        }
+    }
+
+    function isSelectedMenuItem(item) {
+        return menuInfo?.type.name === item?.component?.type.name;
+    }
+
+    function setMenuStyles(item) {
+        let res = [];
+        //close
+        res.push(!openMainMenu && styles.menuItemClose);
+        //open
+        res.push(openMainMenu && styles.menuItemOpen);
+        //selected
+        res.push(isSelectedMenuItem(item) && styles.menuItemSelected);
+
+        return res.join(' ');
+    }
+
+    function setMenuIconStyles(item) {
+        let res = [];
+        //open
+        res.push(openMainMenu && styles.menuItemOpen);
+        //selected
+        res.push(isSelectedMenuItem(item) && styles.menuItemSelected);
+
+        return res.join(' ');
+    }
```

Comment:
Are you sure you really need res.push(false) ?

Possibly better way is:

openMainMenu && res.push(styles.menuItemOpen);

etc